packages/jest-core/src/SearchSource.ts

```diff
@@ -263,7 +263,17 @@ export default class SearchSource {
         changedFiles,
         globalConfig.collectCoverage,
       );
-    } else if (globalConfig.runTestsByPath && paths && paths.length) {
+    }
+
+    let paths = globalConfig.nonFlagArgs;
+
+    if (globalConfig.findRelatedTests && 'win32' === os.platform()) {
+      paths = paths
+        .map(path => glob(path, {nocase: true})[0])
+        .filter(path => path);
+    }
+
+    if (globalConfig.runTestsByPath && paths && paths.length) {
```

Comment:
Lost the elseif here, this code will now run if `globalConfig.onlyChanged && changedFiles`

Reply:
As I see It won't run because there is a `return` statement.

Reply:
You're right, the GH collapse confused me.